sdk/metric/internal/sum_test.go

```diff
@@ -163,6 +164,134 @@ func TestDeltaSumReset(t *testing.T) {
+	metricdatatest.AssertAggregationsEqual(t, want, got, opt)
+
+	// Filtered values should not persist.
+	got = agg.Aggregation()
+	// measured(+): 1, previous(-): 2, filtered(+): 0
+	want.DataPoints = []metricdata.DataPoint[int64]{point[int64](attrs, -1)}
+	metricdatatest.AssertAggregationsEqual(t, want, got, opt)
+	got = agg.Aggregation()
+	// measured(+): 1, previous(-): 1, filtered(+): 0
+	want.DataPoints = []metricdata.DataPoint[int64]{point[int64](attrs, 0)}
+	metricdatatest.AssertAggregationsEqual(t, want, got, opt)
+
+	// Override set value.
+	agg.Aggregate(2, attrs)
+	agg.Aggregate(5, attrs)
+	// Filtered should add.
+	agg.(filterAgg[int64]).filtered(3, attrs)
+	agg.(filterAgg[int64]).filtered(10, attrs)
+	got = agg.Aggregation()
+	// measured(+): 5, previous(-): 1, filtered(+): 13
+	want.DataPoints = []metricdata.DataPoint[int64]{point[int64](attrs, 17)}
+	metricdatatest.AssertAggregationsEqual(t, want, got, opt)
+
+	// Filtered values should not persist.
+	agg.Aggregate(5, attrs)
+	got = agg.Aggregation()
+	// measured(+): 5, previous(-): 18, filtered(+): 0
+	want.DataPoints = []metricdata.DataPoint[int64]{point[int64](attrs, -13)}
+	metricdatatest.AssertAggregationsEqual(t, want, got, opt)
+
+	// Order should not affect measure.
+	// Filtered should add.
+	agg.(filterAgg[int64]).filtered(3, attrs)
+	agg.Aggregate(7, attrs)
+	agg.(filterAgg[int64]).filtered(10, attrs)
+	got = agg.Aggregation()
+	// measured(+): 7, previous(-): 5, filtered(+): 13
+	want.DataPoints = []metricdata.DataPoint[int64]{point[int64](attrs, 15)}
+	metricdatatest.AssertAggregationsEqual(t, want, got, opt)
+	agg.Aggregate(7, attrs)
+	got = agg.Aggregation()
+	// measured(+): 7, previous(-): 20, filtered(+): 0
+	want.DataPoints = []metricdata.DataPoint[int64]{point[int64](attrs, -13)}
+	metricdatatest.AssertAggregationsEqual(t, want, got, opt)
+}
+
+func TestPreComputedCumulativeSum(t *testing.T) {
+	var mono bool
+	agg := NewPrecomputedCumulativeSum[int64](mono)
+	require.Implements(t, (*filterAgg[int64])(nil), agg)
+
+	attrs := attribute.NewSet(attribute.String("key", "val"))
+	agg.Aggregate(1, attrs)
+	got := agg.Aggregation()
+	want := metricdata.Sum[int64]{
+		IsMonotonic: mono,
+		Temporality: metricdata.CumulativeTemporality,
+		DataPoints:  []metricdata.DataPoint[int64]{point[int64](attrs, 1)},
+	}
+	opt := metricdatatest.IgnoreTimestamp()
+	metricdatatest.AssertAggregationsEqual(t, want, got, opt)
+
+	// Cumulative values should persist.
+	got = agg.Aggregation()
+	metricdatatest.AssertAggregationsEqual(t, want, got, opt)
+
+	agg.(filterAgg[int64]).filtered(1, attrs)
+	got = agg.Aggregation()
+	want.DataPoints = []metricdata.DataPoint[int64]{point[int64](attrs, 2)}
+	metricdatatest.AssertAggregationsEqual(t, want, got, opt)
+
+	// Filtered values should not persist.
+	got = agg.Aggregation()
+	want.DataPoints = []metricdata.DataPoint[int64]{point[int64](attrs, 1)}
+	metricdatatest.AssertAggregationsEqual(t, want, got, opt)
```

Comment:
Why is only the filtered dropped?  

Wouldn't this make the first measurement that was recorded have extra meaning?

Reply:
Please see this conversation: https://github.com/open-telemetry/opentelemetry-go/pull/3549#discussion_r1053619102

And to be clear, they are not dropped, they are reset given they were additions to a precomputed sum. The directly set sum, which the user is telling the SDK is the precomputed sum, has semantically different meaning.